### PR TITLE
repo cache

### PR DESCRIPTION
**Pull request by BeanGalSolina**

### CODE REVIEW

File: manager.sh [hunk 1]
```diff
@@ -52,7 +49,25 @@ if [[ "$ACTION" == "get" ]]; then
...
+fi
+
+if [[ "$ACTION" == "update" ]]; then
+  while read -r repo
+  do 
+    REPONAME="$(echo "$repo" | cut -d "=" -f 1)"
+    REPOURL="$(echo "$repo" | cut -d "=" -f 2)"
+    echo "REPO $REPONAME"
+    echo "url $REPOURL"
+    curl "$REPOURL" > "$DIRECTORY/.figmancache/$REPONAME"
```
Comment (PoolloverNathan) [inline manager.sh]: this is one of the seven deadly sins

File: manager.sh [hunk 2]
```diff
@@ -52,7 +49,25 @@ if [[ "$ACTION" == "get" ]]; then
...
   done < "$DIRECTORY/repos.properties"
 fi
 
+if [[ "$ACTION" == "upgrade" ]]; then
+  while read -r file
+  do
+    "$0" get "$(basename "$file" .lua)"
+  done <<< "$(find "$libpath" -type f)"
```
Comment (PoolloverNathan) [inline manager.sh:71]: where does `$libpath` come from??

File: manager.sh [hunk 3]
```diff
@@ -52,7 +49,25 @@ if [[ "$ACTION" == "get" ]]; then
...
+    echo "REPO $REPONAME"
+    echo "url $REPOURL"
+    curl "$REPOURL" > "$DIRECTORY/.figmancache/$REPONAME"
   done < "$DIRECTORY/repos.properties"
 fi
 
+if [[ "$ACTION" == "upgrade" ]]; then
+  while read -r file
+  do
+    "$0" get "$(basename "$file" .lua)"
```
Comment (PoolloverNathan) [inline manager.sh]: bro has never heard of `%%`

File: manager.sh [hunk 4]
```diff
@@ -13,9 +13,7 @@ ACTION="$1"
 ARGUMENT="$2"
 
 source "$DIRECTORY/config.properties"
-
-env
-echo "$libpath"
+mkdir -p "$DIRECTORY/.figmancache"
```
Comment (PoolloverNathan) [inline manager.sh]: what happens when this is in like `/usr/bin/` or `/nix/store/` or something?

**Review by PoolloverNathan — CHANGES_REQUESTED** [review]

File: manager.sh (hunk 1 above)
Comment (BeanGalSolina) [reply manager.sh]: elaborate

Comment (BeanGalSolina) [reply manager.sh]: OH

File: manager.sh (hunk 2 above)
Comment (BeanGalSolina) [reply manager.sh:71]: config

File: manager.sh (hunk 3 above)
Comment (BeanGalSolina) [reply manager.sh]: tf is %%

File: manager.sh (hunk 4 above)
Comment (BeanGalSolina) [reply manager.sh]: figure that out in your nix pr

File: manager.sh (hunk 1 above)
Comment (PoolloverNathan) [reply manager.sh]: > elaborate

you're unconditionally *writing* to a directory next to the binary. i dare you to install this in `/usr/bin` and see what happens.

File: manager.sh (hunk 3 above)
Comment (PoolloverNathan) [reply manager.sh]:
```suggestion
    "$0" get "${file%.lua}"
```

File: manager.sh (hunk 4 above)
Comment (BeanGalSolina) [reply manager.sh]: fixed

Comment (PoolloverNathan) [reply manager.sh]: thanks…

File: manager.sh (hunk 1 above)
Comment (BeanGalSolina) [reply manager.sh]: it doesn't use the binary anymore

Comment (PoolloverNathan) [reply manager.sh]: > it doesn't use the binary anymore
> 

okay that's saner

File: manager.sh (hunk 3 above)
Comment (PoolloverNathan) [reply manager.sh]: oh yeah if you're relying on basename stripping directory components this won't work

**Review by PoolloverNathan — APPROVED** [review]
yolo

**Pull request merged**